CSSGapDecorations/explainer.md

```diff
@@ -456,6 +457,42 @@ grid lines, and (b) simplifying the model for fine-tuning segment placement. We
 also believe the proposal in this explainer offers developers more flexibility
 even absent support for gap decoration images; see Scenario 3 for one example.
 
+### Alternative 2: Using pseudo-elements
+
+An alternative approach to `column-rule-*` and `row-rule-*` properties would be
+to introduce pseudo-elements representing gaps, for example:
+
+```css
+.container {
+  display: grid;
+  grid-template: auto / auto;
+  gap: 5px;
+}
+.container::column-gaps {
+  background: red;
+  width: 1px;
+}
+.container::row-gaps {
+  background: blue;
+  width: 1px;
+}
+```
+
+In some ways, this would be more powerful, as it would allow for more
+flexibility for what can be placed in the gaps.
+
+However, this approach also comes with drawbacks.
+Varying gap decorations over a container becomes much harder. One might
+imagine a `::row-gaps::nth(even)` pseudo selector to style every other row
+gap. However, certain container types such as grid can automatically generate
+rows and columns depending on their contents. That means we don't know until
+layout time how many such pseudo styles we need to produce, which creates a
+wrong-way dependency between layout and style. It would also mean that, for
+large containers, we would incur the costs of calculating styles for every
```

Comment:
I think the other point here was that we would need objects for all the gaps (which I guess could be implied by the need to calculate styles for every gap)

Reply:
Thanks, I changed "calculating styles" to "calculating and storing styles" to capture this point.